pkg/controller/logstorage/logstorage.go

```diff
@@ -47,7 +47,7 @@ func (r *ReconcileLogStorage) createLogStorage(
 	hdler utils.ComponentHandler,
 	reqLogger logr.Logger,
 	ctx context.Context,
-) (reconcile.Result, bool, error) {
+) (reconcile.Result, bool, bool, error) {
```

Comment:
Could you add a function definition to this, especially to include comments on what the return values are?

Reply:
Definition added:
```diff
+// createLogStorage Is called by Reconcile() in the Logstorage controller to render its components
+//
+// It returns 4 values:
+// A reconcile.Result to be returned by Reconcile() if the 'proceed' bool is false or if there is an error,
+// a 'proceed' bool indicating if the Reconcile function should proceed with the reconcile process,
+// a 'finalizerCleanup' bool indicating if it's safe to remove the LogStorage finalizer (only on deletion and after
+// kibana and elasticsearch components were successfully removed), and lastly,
+// an error if there was any or nil otherwise
```